index.js

```diff
@@ -270,35 +287,78 @@ class ExecutorQueue extends Executor {
         return this.redisBreaker.runCommand('hdel', this.periodicBuildTable, config.jobId);
     }
 
+    /**
+     * Calls postBuildEvent() with job configuration
+     * @async _startNewEvent
+     * @param {Object} config       Configuration
+     * @return {Promise}
+     */
+    async _startNewEvent(config) {
+        return this.postBuildEvent(config)
+            .catch(() => Promise.resolve());
+    }
+
     /**
      * Starts a new build in an executor
      * @async  _start
      * @param  {Object} config               Configuration
      * @param  {Object} [config.annotations] Optional key/value object
      * @param  {String} config.build         Build object
      * @param  {Array}  config.blockedBy     Array of job IDs that this job is blocked by. Always blockedby itself
+     * @param  {Array}  config.freezeWindows Array of cron expressions that this job cannot run during
      * @param  {String} config.apiUri        Screwdriver's API
      * @param  {String} config.jobId         JobID that this build belongs to
      * @param  {String} config.buildId       Unique ID for a build
+     * @param  {String} config.eventId       eventID that this build belongs to
+     * @param  {Object} config.pipeline      Pipeline of the job
+     * @param  {Fn}     config.tokenGen      Function to generate JWT from username, scope and scmContext
      * @param  {String} config.container     Container for the build to run in
      * @param  {String} config.token         JWT to act on behalf of the build
      * @return {Promise}
      */
     async _start(config) {
         await this.connect();
-        const { build, buildId, jobId, blockedBy } = config;
+        const { build, buildId, jobId, blockedBy, freezeWindows } = config;
+
+        if (!this.tokenGen) {
+            this.tokenGen = config.tokenGen;
+        }
 
         delete config.build;
-        // Store the config in redis
-        await this.redisBreaker.runCommand('hset', this.buildConfigTable,
-            buildId, JSON.stringify(config));
 
-        // Note: arguments to enqueue are [queue name, job name, array of args]
-        const enq = await this.queueBreaker.runCommand('enqueue', this.buildQueue, 'start', [{
-            buildId,
-            jobId,
-            blockedBy: blockedBy.toString()
-        }]);
+        const currentTime = new Date();
+        const origTime = new Date(currentTime.getTime());
+
+        timeOutOfWindows(freezeWindows, currentTime);
```

Comment:
does this need to be reassigned?
`currentTime = timeOutOfWindows(freezeWindows, currentTime)`

Reply:
Does not need to be reassigned, date operations are destructive

Reply:
Can we rename the variable to something else?
The if condition seems confusing `(if currentTime > origTime)` sounds strange... lol